collated/main/tby1994.md

```diff
@@ -0,0 +1,884 @@
+
+        CommandBox commandBox = new CommandBox(logic);
+        commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
+    }
+```
+###### /java/seedu/address/ui/TaskCard.java
+``` java
+/**
+ * An UI component that displays information of a {@code Task}.
+ */
+public class TaskCard extends UiPart<Region> {
+
+    private static final String FXML = "TaskCard.fxml";
+    private static final int GREEN_RANGE = 7;
+    private static final int YELLOW_RANGE = 3;
+    private static final int RED_RANGE = 0;
+    private static String[] colours = { "red", "orange", "cyan", "green", "blue", "purple", "pink", "grey", "black" };
+    private static HashMap<String, String> tagColours = new HashMap<String, String>();
+    private static Random random = new Random();
+
+    public final ReadOnlyTask task;
+
+    @FXML
+    private GridPane gridPane;
+
+    @FXML
+    private Label id;
+
+    @FXML
+    private Label description;
+
+    @FXML
+    private Label startDate;
+
+    @FXML
+    private Label deadline;
+
+    @FXML
+    private FlowPane tags;
+
+    public TaskCard(ReadOnlyTask task, int displayedIndex) {
+        super(FXML);
+        this.task = task;
+        id.setText("  " + displayedIndex + ". ");
+        initTags(task);
+        bindListeners(task);
+    }
+
+    /**
+     * Binds the individual UI elements to observe their respective {@code Task} properties
+     * so that they will be notified of any changes.
+     */
+    private void bindListeners(ReadOnlyTask task) {
+        description.textProperty().bind(Bindings.convert(task.descriptionProperty()));
+        startDate.textProperty().bind(Bindings.convert(task.startDateProperty()));
+        deadline.textProperty().bind(Bindings.convert(task.deadlineProperty()));
+        setColour();
+    }
+```
+###### /java/seedu/address/ui/TaskCard.java
+``` java
+    /**
+     * Set the colour of label for the same tag
+     * @param task
+     */
+    private void initTags(ReadOnlyTask task) {
+        task.getTags().forEach(tag -> {
+            Label tagLabel = new Label(tag.tagName);
+            tagLabel.setStyle("-fx-background-color: " + getColourForTag(tag.tagName));
+            tags.getChildren().add(tagLabel);
+        });
+    }
+
+    private static String getColourForTag(String tagValue) {
+        if (!tagColours.containsKey(tagValue)) {
+            tagColours.put(tagValue, colours[random.nextInt(colours.length)]);
+        }
+
+        return tagColours.get(tagValue);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof TaskCard)) {
+            return false;
+        }
+
+        // state check
+        TaskCard card = (TaskCard) other;
+        return id.getText().equals(card.id.getText())
+            && task.equals(card.task);
```

Comment:
this line wrap doesn't seems necessary, also it should be 8 indentation spaces if its intended.